SpatialGDK/Source/SpatialGDK/Private/Interop/AsyncPackageLoadFilter.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/AsyncPackageLoadFilter.h"
+
+#include "UObject/UObjectGlobals.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+
+DEFINE_LOG_CATEGORY(LogAsyncPackageLoadFilter);
+
+namespace SpatialGDK
+{
+AsyncPackageLoadFilter::AsyncPackageLoadFilter(USpatialNetDriver* InNetDriver)
+	: NetDriver(InNetDriver)
+{
+}
+
+bool AsyncPackageLoadFilter::ProcessAndCheckEntityFilter(Worker_EntityId EntityId, const FString& ClassPath)
+{
+	if (IsEntityWaitingForAsyncLoad(EntityId))
+	{
+		return true;
+	}
+
+	if (NeedToLoadClass(ClassPath))
+	{
+		StartAsyncLoadingClass(EntityId, ClassPath);
+		return true;
+	}
+
+	return false;
+}
+
+bool AsyncPackageLoadFilter::NeedToLoadClass(const FString& ClassPath)
+{
+	UObject* ClassObject = FindObject<UClass>(nullptr, *ClassPath, false);
+	if (ClassObject == nullptr)
+	{
+		return true;
+	}
+
+	FString PackagePath = GetPackagePath(ClassPath);
+	FName PackagePathName = *PackagePath;
+
+	// UNR-3320 The following test checks if the package is currently being processed in the async loading thread.
```

Comment:
are these tickets duplicates?

Reply:
This is from the old async loading code, will change the comment to indicate the ticket refers to the context, not the open issue.